sound/soc/soc-dapm.c

```diff
@@ -4464,9 +4464,27 @@ void snd_soc_dapm_connect_dai_link_widgets(struct snd_soc_card *card)
 			for_each_rtd_codec_dais(rtd, i, codec_dai)
 				dapm_connect_dai_pair(card, rtd, codec_dai,
 						      asoc_rtd_to_cpu(rtd, i));
+		} else if (rtd->dai_link->num_codecs > rtd->dai_link->num_cpus &&
+			   rtd->dai_link->codec_ch_maps) {
+			int cpu_id;
+
+			for_each_rtd_codec_dais(rtd, i, codec_dai) {
+				cpu_id = rtd->dai_link->codec_ch_maps[i].connected_cpu_id;
+				if (cpu_id >= rtd->dai_link->num_cpus) {
+					dev_err(card->dev,
+						"cpu_id %d is out of range\n", cpu_id);
```

Comment:
"cpu_id %d too large, num_cpus is %d\n", cpu_id, rtd->dai_link->num_cpus);

Reply:
along with the dai_link name maybe to tell which dai_link is in violation?

Reply:
@plbossart @ranj063 Change to 
```
dev_err(card->dev,
        "dai_link %s cpu_id %d too large, num_cpus is %d\n",
        rtd->dai_link->name, cpu_id,
        rtd->dai_link->num_cpus);
```